### PR TITLE
Enable Conditional Publish Behaviors for Azure and MyGet storage

### PR DESCRIPTION
**Pull request by MattGal**
**Note: As we are trying very hard to not break the release builds this week I have marked with * NO MERGE ***, this is just showing how it would be done and make it possible to get rid of extra branches which exist just to do this.

Introduces two new build definition variables, PublishToBlobStorage and PublishToMyGet, and makes various steps that do these sorts of publishes conditional on them.    See [the VSTS docs](https://docs.microsoft.com/en-us/vsts/build-release/concepts/process/conditions#examples) on Conditional Task Expressions for more info on this.

In addition to these changes merging, we'd also need to do the following:

- add "-b 3.2" to update the API Version used to talk to the VSTS API when hydrating build definitions.
- Add PublishToBlobStorage and PublishToMyGet variables to the Pipebuild defintion used to kick off the build.

### CODE REVIEW

File: buildpipeline/DotNet-CoreFx-Trusted-OSX.json [hunk 1]
```diff
@@ -185,6 +185,7 @@
       "alwaysRun": false,
       "displayName": "Push packages to Azure",
       "timeoutInMinutes": 0,
+      "condition": "and(succeeded(), eq(variables['PublishToBlobStorage'], 'True'))",
```
Comment (weshaggard) [inline buildpipeline/DotNet-CoreFx-Trusted-OSX.json]: I don't think the individual legs need to be conditioned just the publish step in the publish leg.

File: buildpipeline/DotNet-Trusted-Publish.json [hunk 2]
```diff
@@ -145,6 +146,7 @@
       "alwaysRun": false,
```
Comment (weshaggard) [inline buildpipeline/DotNet-Trusted-Publish.json]: I expect to see a publish to blob feed step in this definition.

File: buildpipeline/DotNet-CoreFx-Trusted-OSX.json (hunk 1 above)
Comment (MattGal) [reply buildpipeline/DotNet-CoreFx-Trusted-OSX.json]: This condition only disables this specific step (invocation leading to copying packages to azure blob storage).  We currently do all the "publish things to Azure storage" stuff in the individual legs, would you prefer I pull that out?   

While I get that we don't need this right now, it defaults to True (so won't disable anything) and leaves something behind that allows us to test build processes without filling up storage accounts.

File: buildpipeline/DotNet-Trusted-Publish.json (hunk 2 above)
Comment (MattGal) [reply buildpipeline/DotNet-Trusted-Publish.json]: Reviewing what was already there, all blob uploading I saw for this branch was done in the individual legs as noted above.

Let's chat about this tomorrow AM as I obviously can't merge this today regardless.

Comment (MattGal) [reply buildpipeline/DotNet-Trusted-Publish.json]: Updated to allow all legs to continue uploading built packages to blob storage, but left (only) the PublishToBlobStorage variable as a place holder to use when transport feeds turn on.

Comment (MattGal) [reply buildpipeline/DotNet-Trusted-Publish.json]: I'll synch with @karajas  when I can about where that work is.  Currently there is no publish-to-blob-feed in release/2.0.0

Comment (chcosta) [reply buildpipeline/DotNet-Trusted-Publish.json]: nit: Leaving the PublishToBlobStorage variable is nice, but somewhat confusing until it's implemented.  Also, it pre-supposes the implementation for transport feeds is going to follow your mental model if/when it's implemented.  I'd remove it and file an issue assigned to Karthik or JC to make sure it happens.  That's my 2 cents.

File: buildpipeline/DotNet-Trusted-Publish-Symbols.json [hunk 3]
```diff
@@ -46,6 +46,7 @@
       "alwaysRun": false,
       "displayName": "Extract symbol packages; if release branch, archive",
       "timeoutInMinutes": 0,
+      "condition": "and(succeeded(), eq(variables['PublishToMyGet'], 'True'))",
```
Comment (chcosta) [inline buildpipeline/DotNet-Trusted-Publish-Symbols.json]: I think that the only reason this step is an inline powershell step is because Conditional's were not supported.  From a debugging / reading / simplicity perspective, it'd be much nicer if this was just the `build-managed.cmd -- blah` part and utilized the conditional.

Comment (MattGal) [reply buildpipeline/DotNet-Trusted-Publish-Symbols.json]: I agree it's not pretty, but the string methods exposed to Conditional Tasks don't seem to include something like startsWith, so unless you know that's defined I'd rather not guess at this and be wrong.

File: buildpipeline/DotNet-Trusted-Publish-Symbols.json [hunk 4]
```diff
@@ -46,6 +46,7 @@
       "alwaysRun": false,
       "displayName": "Extract symbol packages; if release branch, archive",
```
Comment (weshaggard) [inline buildpipeline/DotNet-Trusted-Publish-Symbols.json]: How is this related to myget? I don't see us pushing anything to myget in this step unless it is happening under the covers.

Comment (MattGal) [reply buildpipeline/DotNet-Trusted-Publish-Symbols.json]: it's related in that if we don't publish to myget but we publish the symbols to MS symbol servers we'll have posted symbols that end-users can't get to.  Mostly, the idea is to not waste time and resources with symbols noone is likely to need, since the flat blob storage will have symbol packages included.

Should I remove it?

Comment (weshaggard) [reply buildpipeline/DotNet-Trusted-Publish-Symbols.json]: I don't know that I would condition this on the myget publish condition, as I think we would want to do this if we do any sort of publish. If we find it interesting we could have a NoPublicPublish option or something that could cover this if we need.

**Comment by MattGal** [comment]
Closing, already integrated with @karajas 's work
